jetty-servlets/src/main/java/org/eclipse/jetty/servlets/DoSFilter.java

```diff
@@ -305,67 +307,76 @@ protected void doFilter(HttpServletRequest request, HttpServletResponse response
 
         // Look for the rate tracker for this request.
         RateTracker tracker = (RateTracker)request.getAttribute(__TRACKER);
-        if (tracker == null)
+        if (tracker != null)
+        {
+            // Redispatched, RateTracker present in request attributes.
+            throttleRequest(request, response, filterChain, tracker);
```

Comment:
This looks strange.... I know it is the same behaviour as before, but now you have restructured it, I think it  is just wrong to always throttle async dispatched requests. 

I also think it is strange that we can only have a single tracker per request: session or IP or UserID.   We should probably track all three for a single request... but maybe that is another PR... and maybe in 10.